src/test/java/com/google/cloud/compute/v1/it/ITInstanceTest.java

```diff
@@ -42,42 +76,191 @@ public static void setUp() throws IOException {
+                    .setDescription("updated description")
+                    .setFingerprint(instance.getFingerprint())
+                    .build())
+            .addAllFieldMask(fieldMask)
             .build();
+    if (instance.getStatus().equals("RUNNING")) {
+      instanceClient.updateInstance(request);
+      Instance updatedInstance = getInstance();
+      assertThat(updatedInstance.getDescription()).isNotEqualTo(instance.getDescription());
+      assertThat(updatedInstance.getDescription()).isEqualTo("updated description");
+    }
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void instanceStartStopTest() {
+    StopInstanceHttpRequest stopInstanceHttpRequest =
+        StopInstanceHttpRequest.newBuilder().setInstance(FORMATTED_INSTANCE).build();
+    Operation stopOperation = instanceClient.stopInstance(stopInstanceHttpRequest);
+    assertThat(stopOperation.getOperationType()).isEqualTo("stop");
+
+    StartInstanceHttpRequest startInstanceHttpRequest =
+        StartInstanceHttpRequest.newBuilder().setInstance(FORMATTED_INSTANCE).build();
+    Operation startOperation = instanceClient.startInstance(startInstanceHttpRequest);
+    assertThat(startOperation.getOperationType()).isEqualTo("start");
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void setSchedulingInstanceTest() {
+    ShieldedInstanceIntegrityPolicy shieldedInstanceIntegrityPolicyResource =
+        ShieldedInstanceIntegrityPolicy.newBuilder().setUpdateAutoLearnPolicy(true).build();
+    List<String> fieldMask = new ArrayList<>();
     try {
-      instanceClient.insertInstance(ProjectZoneName.of(DEFAULT_PROJECT, ZONE), instance);
-    } catch (InvalidArgumentException e) {
-      // Expect a Bad Request HTTP 400 error, but it should NOT be because of a resource name
-      // problem.
-      assertThat(e.getMessage()).contains("Bad Request");
-      assertThat(e.getCause().getMessage())
-          .contains("Invalid value for field 'resource.networkInterfaces'");
-      return;
+      Operation operation =
+          instanceClient.setShieldedInstanceIntegrityPolicyInstance(
+              FORMATTED_INSTANCE, shieldedInstanceIntegrityPolicyResource, fieldMask);
+      Assert.fail();
+    } catch (UnknownException ex) {
+      assertThat(ex.getCause().getMessage()).isEqualTo("HTTP method PATCH not supported");
     }
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void simulateMaintenanceEventInstanceTest() {
+    Operation operation = instanceClient.simulateMaintenanceEventInstance(FORMATTED_INSTANCE);
+    assertOperationDetails(operation, "simulateMaintenanceEvent");
+  }
+
+  @Test
+  @SuppressWarnings("all")
+  public void setDeletionProtectionInstanceTest() {
+    ProjectZoneInstanceResourceName resource =
+        ProjectZoneInstanceResourceName.of(DEFAULT_PROJECT, INSTANCE, ZONE);
+    Operation operation = instanceClient.setDeletionProtectionInstance(resource, false);
+    assertOperationDetails(operation, "setDeletionProtection");
+  }
```

Comment:
Here and below, the tests which perform some operation should check not only the returned operation object, but the fact that operation affects the instance as expected.

Reply:
done